### ISSUE TITLE
Invalid SQL for MySQL 5.7 when using union in subqueries

### ISSUE THREAD

**Issue by rolfvandekrol** [laravel/framework#53555]
### Laravel Version

11.32.0

### PHP Version

8.3

### Database Driver & Version

MySQL 5.7

### Description

I encountered a situation where Laravel generates SQL that is valid in MySQL 8 and not valid in MySQL 5.7. I'm not 100% sure this is really a bug. Since it works in MySQL 8 and it is a bit of an edge case. However, MySQL 5.7 is officially supported AFAIK.

When generating select queries with union, Laravel adds parentheses around the individual queries. This works fine when the query is used directly, but not when the query is used as a subquery.

See also https://dev.mysql.com/doc/refman/8.0/en/union.html for more information on the difference between union handling between MySQL 5.7 and 8.

### Steps To Reproduce

I made a example repo to demonstrate the problem. https://github.com/rolfvandekrol/laravel-mysql-union

In this small app, there are 3 models. Foo, Bar and Baz. Bar and Baz both have a property `b` and belong to Foo. I want to retrieve all Foos, with the lowest value of `b` for the underlying Bar and Baz instances.

To do this, I create query like this (also in the `app:qux` command in the example repo).

```php
        $bQuery =
            Bar::select('b')
                ->whereColumn('foo_id', 'foos.id')
                ->union(
                    Baz::select('b')
                        ->whereColumn('foo_id', 'foos.id')
                )
                ->orderBy('b')
                ->limit(1);

        $query = Foo::select([
            '*',
            'b' => $bQuery,
        ]);
```

This generates a query like this:

```sql
select *, ((select `b` from `bars` where `foo_id` = `foos`.`id`) union (select `b` from `bazs` where `foo_id` = `foos`.`id`) order by `b` asc limit 1) as `b` from `foos`
```

This fails on MySQL 5.7 with the following error:

```
SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'union (select `b` from `bazs` where `foo_id` = `foos`.`id`) order by `b` asc lim' at line 1 (Connection: mysql, SQL: select *, ((select `b` from `bars` where `foo_id` = `foos`.`id`) union (select `b` from `bazs` where `foo_id` = `foos`.`id`) order by `b` asc limit 1) as `b` from `foos`)
```

On MySQL 8, it works fine. 

To make a query that would work in both 5.7 and 8, it should be this (notice the missing parentheses)

```sql
select *, (select `b` from `bars` where `foo_id` = `foos`.`id` union select `b` from `bazs` where `foo_id` = `foos`.`id` order by `b` asc limit 1) as `b` from `foos`
```

**Comment by crynobone** [comment]
Hey there,

While this may be a legitimate issue, can you first try posting your problem or question on one of the support channels below? If this issue can be definitively identified as a bug, feel free to open up a new issue with a link to the original one and we'll gladly help you out.

* [Laracasts Forums](https://laracasts.com/discuss)
* [Laravel.io Forums](https://laravel.io/forum)
* [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
* [Discord](https://discordapp.com/invite/KxwQuKb)
* [Larachat](https://larachat.co/)
* [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)

Thanks!

**Comment by rolfvandekrol** [comment]
@crynobone The doubt I expressed in my introduction wasn't whether i did encounter a legitimate bug. The doubt was whether is was worth fixing. Which is not something that can be resolved in a support channel. This is ultimately a matter of the Laravel team deciding whether a bug is impactful enough to warrant their time. I simply switched to MySQL 8 to resolve this. Maybe this needs to be documented somewhere, that unions in subqueries are not working in MySQL 5.7.
